cmd/credential.go

```diff
@@ -75,65 +58,71 @@ type credential struct {
 	secretKeyHash []byte
 }
 
-// Generate a bcrypt hashed key for input secret key.
-func mustGetHashedSecretKey(secretKey string) []byte {
-	hashedSecretKey, err := bcrypt.GenerateFromPassword([]byte(secretKey), bcrypt.DefaultCost)
-	if err != nil {
-		console.Fatalf("Unable to generate secret hash for secret key. Err: %s.\n", err)
-	}
-	return hashedSecretKey
+// IsValid - returns whether credential is valid or not.
+func (cred credential) IsValid() bool {
+	return isAccessKeyValid(cred.AccessKey) && isSecretKeyValid(cred.SecretKey)
 }
 
-// Initialize a new credential object
-func newCredential() credential {
-	return newCredentialWithKeys(mustGetAccessKey(), mustGetSecretKey())
-}
+// Equals - returns whether two credentials are equal or not.
+func (cred credential) Equal(ccred credential) bool {
+	if !ccred.IsValid() {
```

Comment:
@krishnasrinivas I am unable to understand what code flow you are asking.

This method checks for equality of two credentials.  Those credentials could be created with different method than `createCredential()` or `mustGetNewCredential()`.  `credential.Equal()` prevents such error case by validating passed `ccred` is valid or not.

In short, if both `cred` and `ccred` are empty, this method makes equality check failure.